shiny/ui/_include_helpers.py

```diff
@@ -0,0 +1,300 @@
+            same directory as ``path`` that contain sensitive information**. A good
+            general rule of thumb to follow is to have ``path`` be located in a
+            subdirectory of the app directory. For example, if the app's source is
+            located at ``/app/app.py``, then ``path`` should be somewhere like
+            ``/app/html/index.html`` (and all the other relevant accompanying 'safe'
+            files should be located under ``/app/html/``).
+    attrs
+        Additional attributes to add to the :func:`~ui.tags.iframe` tag.
+
+    Returns
+    -------
+    A :func:`~ui.tags.iframe` tag.
+
+    Note
+    ----
+    For safety reasons, this function includes the HTML file as a
+    :func:`~ui.tags.iframe`, which means it's 'isolated' from the rest of the parent
+    document. If instead, you don't want to isolate (meaning, among other things, you
+    want the HTML inherit CSS styles from the parent document), you can do something
+    like this:
+
+    .. code-block:: python
+        from shiny import ui
+
+        with open("custom.html", "r", encoding="utf-8") as f:
+            custom_html = ui.HTML(f.read())
+
+        app_ui = ui.page_fluid(..., custom_html, ...)
+
+    See also
+    --------
+    ~ui.tags.iframe
+    ~ui.HTML
+    ~include_javascript
+    ~include_css
+    """
+
+    include_files = method == "link_files"
+    path_dest, hash = maybe_copy_files(path, include_files)
+
+    dep, src = create_include_dependency(
+        "include-html-" + hash, path_dest, include_files
+    )
+
+    default_attrs: Dict[str, TagAttrArg] = {
+        "src": src,
+        "scrolling": "no",
+        "seamless": "seamless",
+        "frameBorder": "0",
+    }
+
+    return tags.iframe(default_attrs, dep, **attrs)
+
+
+# ---------------------------------------------------------------------------
+# Include helpers
+# ---------------------------------------------------------------------------
+
+
+def create_include_dependency(
+    name: str, path: str, include_files: bool
+) -> Tuple[HTMLDependency, str]:
+    dep = HTMLDependency(
+        name,
+        DEFAULT_VERSION,
+        source={"subdir": os.path.dirname(path)},
+        all_files=include_files,
+    )
+
+    # source_path_map() tells us where the source subdir is mapped to on the client
+    # (i.e., session._register_web_dependency() uses the same thing to determine where
+    # to mount the subdir, but we can't assume an active session at this point).
+    src = os.path.join(dep.source_path_map()["href"], os.path.basename(path))
+
+    return dep, src
+
+
+def maybe_copy_files(path: str, include_files: bool) -> Tuple[str, str]:
+    hash = get_hash(path, include_files)
+
+    # To avoid unnecessary work when the same file is included multiple times,
+    # use a directory scoped by a hash of the file.
+    tmpdir = os.path.join(tempfile.gettempdir(), "shiny_include_files", hash)
+    path_dest = os.path.join(tmpdir, os.path.basename(path))
```

Comment:
I find it a bit concerning that we copy files to a temp dir and there's no provision for removing them after app stops or when Python exits... but then again, I see a lot of junk in my temp dir that has been there for a while, so maybe this isn't so bad.

Reply:
Also these files are probably going to be pretty small relative to what goes into temp